subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherUpdater.java

```diff
@@ -17,13 +17,13 @@
 package org.gradle.internal.watch.registry;
 
 import org.gradle.internal.snapshot.CompleteFileSystemLocationSnapshot;
-import org.gradle.internal.snapshot.SnapshotHierarchy;
+import org.gradle.internal.vfs.SnapshotDiffListener;
 import org.gradle.internal.watch.WatchingNotSupportedException;
 
 import java.io.File;
 import java.util.Collection;
 
-public interface FileWatcherUpdater extends SnapshotHierarchy.SnapshotDiffListener {
+public interface FileWatcherUpdater extends SnapshotDiffListener {
```

Comment:
I think this shouldn't implement the listener. We should instead create a separate listener that calls this thing. I think we are actually doing that already, so this here is just an oversight.